packages/component-library-angular/src/button-link/ButtonLink.spec.ts

```diff
@@ -0,0 +1,190 @@
+import { render, screen } from '@testing-library/angular';
+import '@testing-library/jest-dom';
+import { UtrechtButtonLinkAttr } from './component';
+
+describe('Button Link', () => {
+  it('renders a button link role element', async () => {
+    await render('<a utrecht-button-link href="https://example.com">Link</a>', {
+      declarations: [UtrechtButtonLinkAttr],
+    });
+
+    const buttonLink = screen.getByRole('link', { name: 'Link' });
+
+    expect(buttonLink).toBeInTheDocument();
+    expect(buttonLink).toBeVisible();
+  });
+
+  it('renders an HTML button link element', async () => {
+    const { container } = await render('<a utrecht-button>Read more</a>', {
+      declarations: [UtrechtButtonLinkAttr],
+    });
+
+    const buttonLink = container.querySelector('a:only-child');
+
+    expect(buttonLink).toBeInTheDocument();
+  });
+
+  it('renders button link that contain HTML rich text content', async () => {
+    const { container } = await render(
+      `<a href="https://example.com" utrecht-button-link>
+          Visit <strong>now</strong>
+        </a>`,
+      {
+        declarations: [UtrechtButtonLinkAttr],
+      },
+    );
+    const buttonLink = container.querySelector(':only-child');
+
+    const richText = buttonLink?.querySelector('strong');
+
+    expect(buttonLink).toContainElement(richText as HTMLElement);
+  });
+
+  it('renders a design system BEM class name', async () => {
+    const { container } = await render('<a href="https://example.com" utrecht-button-link></a>', {
+      declarations: [UtrechtButtonLinkAttr],
+    });
+
+    const buttonLink = container.querySelector(':only-child');
+
+    expect(buttonLink).toHaveClass('utrecht-button-link');
+    expect(buttonLink).toHaveClass('utrecht-button-link--html-a');
+  });
+
+  it('can be hidden', async () => {
+    const { container } = await render('<a href="https://example.com" [hidden]="hidden" utrecht-button-link></a>', {
+      declarations: [UtrechtButtonLinkAttr],
+      componentProperties: {
+        hidden: true,
+      },
+    });
+
+    const buttonLink = container.querySelector(':only-child');
+
+    expect(buttonLink).not.toBeVisible();
+  });
+
+  it('can have a addition class name', async () => {
+    const { container } = await render('<a href="https://example.com" [class]="class" utrecht-button-link></a>', {
+      declarations: [UtrechtButtonLinkAttr],
+      componentProperties: {
+        class: 'custom-class',
+      },
+    });
+
+    const buttonLink = container.querySelector(':only-child');
+
+    expect(buttonLink).toHaveClass('custom-class');
+    expect(buttonLink).toHaveClass('utrecht-button-link');
+    expect(buttonLink).toHaveClass('utrecht-button-link--html-a');
```

Comment:
I don't think you need to test for `utrecht-button-link--html-a`, since I don't consider it to be a public API. We should be able to refactor the CSS selector from `utrecht-button-link--html-a` to `a.utrecht-button-link` without anyones code breaking. But we can never change `utrecht-button-link` that way.